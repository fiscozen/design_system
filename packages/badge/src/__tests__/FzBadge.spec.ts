import { describe, it } from "vitest";
import { mount } from "@vue/test-utils";
import { FzBadge } from "..";

describe.concurrent("FzBadge", () => {
  it("black color matches snapshot", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      props: {
        color: "black",
      },
      slots: {
        default: "Fiscozen",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("error color matches snapshot", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      props: {
        color: "error",
      },
      slots: {
        default: "Fiscozen",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("warning color matches snapshot", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      props: {
        color: "warning",
      },
      slots: {
        default: "Fiscozen",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("success color matches snapshot", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      props: {
        color: "success",
      },
      slots: {
        default: "Fiscozen",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("info color matches snapshot", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      props: {
        color: "info",
      },
      slots: {
        default: "Fiscozen",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should have rounded-full class when default slot is a string with 1 character", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      slots: {
        default: "1",
      },
    });

    expect(wrapper.classes()).toContain("rounded-full");
  });

  it("should not have rounded-full class when default slot is a string with more than 1 character", async ({ expect }) => {
    const wrapper = mount(FzBadge, {
      slots: {
        default: "Fiscozen",
      },
    });

    expect(wrapper.classes()).not.toContain("rounded-full");
  });
});
