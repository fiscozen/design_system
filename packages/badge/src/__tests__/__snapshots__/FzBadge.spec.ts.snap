// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzBadge > black color matches snapshot 1`] = `"<div class="text-xs px-12 rounded-xl w-fit bg-core-black text-core-white">Fiscozen</div>"`;

exports[`FzBadge > error color matches snapshot 1`] = `"<div class="text-xs px-12 rounded-xl w-fit bg-semantic-error text-core-white">Fiscozen</div>"`;

exports[`FzBadge > info color matches snapshot 1`] = `"<div class="text-xs px-12 rounded-xl w-fit bg-semantic-info text-core-white">Fiscozen</div>"`;

exports[`FzBadge > success color matches snapshot 1`] = `"<div class="text-xs px-12 rounded-xl w-fit bg-semantic-success text-core-white">Fiscozen</div>"`;

exports[`FzBadge > warning color matches snapshot 1`] = `"<div class="text-xs px-12 rounded-xl w-fit bg-semantic-warning">Fiscozen</div>"`;
